components/raftstore/src/store/peer.rs

```diff
@@ -176,6 +180,143 @@ pub struct CheckTickResult {
     up_to_date: bool,
 }
 
+pub struct ProposedAdminCmd<S>
+where
+    S: Snapshot,
+{
+    epoch_state: AdminCmdEpochState,
+    index: u64,
+    cbs: Vec<Callback<S>>,
+}
+
+impl<S: Snapshot> ProposedAdminCmd<S> {
+    fn new(epoch_state: AdminCmdEpochState, index: u64) -> ProposedAdminCmd<S> {
+        ProposedAdminCmd {
+            epoch_state,
+            index,
+            cbs: Vec::new(),
+        }
+    }
+}
+
+struct CmdEpochChecker<S>
+where
+    S: Snapshot,
+{
+    // Although it's a deque, because of the characteristics of the settings from `ADMIN_CMD_EPOCH_MAP`,
+    // the max size of admin cmd is 2, i.e. split/merge and change peer.
+    proposed_admin_cmd: VecDeque<ProposedAdminCmd<S>>,
+    term: u64,
```

Comment:
Could we also have a `current_epoch` here (it can be updated in `advance_apply`) ? to check epoch when `proposed_admin_cmd` is cleared, it will be useful for stale messages.